src/test/java/redis/clients/jedis/tests/commands/SetCommandsTest.java

```diff
@@ -142,34 +145,46 @@ public void spop() {
   public void spopWithCount() {
     jedis.sadd("foo", "a");
     jedis.sadd("foo", "b");
+    jedis.sadd("foo", "c");
 
-    Set<String> expected = new HashSet<String>();
-    expected.add("a");
-    expected.add("b");
+    Set<String> superSet = new HashSet<String>();
+    superSet.add("c");
+    superSet.add("b");
+    superSet.add("a");
 
     Set<String> members = jedis.spop("foo", 2);
 
     assertEquals(2, members.size());
-    assertEquals(expected, members);
+    assertCollectionContainsAll(superSet, members);
+    superSet.removeAll(members);
 
     members = jedis.spop("foo", 2);
-    assertTrue(members.isEmpty());
+    assertEquals(1, members.size());
+    assertEquals(superSet, members);
+
+    assertNull(jedis.spop("foo", 2));
 
     // Binary
     jedis.sadd(bfoo, ba);
     jedis.sadd(bfoo, bb);
+    jedis.sadd(bfoo, bc);
 
-    Set<byte[]> bexpected = new HashSet<byte[]>();
-    bexpected.add(bb);
-    bexpected.add(ba);
+    Set<byte[]> bsuperSet = new HashSet<byte[]>();
+    bsuperSet.add(bc);
+    bsuperSet.add(bb);
+    bsuperSet.add(ba);
 
     Set<byte[]> bmembers = jedis.spop(bfoo, 2);
 
     assertEquals(2, bmembers.size());
-    assertByteArraySetEquals(bexpected, bmembers);
+    assertByteArrayCollectionContainsAll(bsuperSet, bmembers);
```

Comment:
Why not having a simple comparison and make sure we're getting back the exact two elements we expect?